File: rb/features.rb
# Nominatim SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module NominatimFeatures
  def self.make_feature(name)
    case name
    when "base"
      NominatimBaseFeature.new
    when "test"
      NominatimTestFeature.new
    else
      NominatimBaseFeature.new
    end
  end
end

File: py/features.py
# Nominatim SDK feature factory

from feature.base_feature import NominatimBaseFeature
from feature.test_feature import NominatimTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NominatimBaseFeature(),
        "test": lambda: NominatimTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
